dnsmasq/Makefile

```diff
@@ -20,20 +20,77 @@
 #   [TAG=1.0] [REGISTRY=gcr.io/google_containers] [NAMESERVER=127.0.0.1#10053] make container
 
 # Default registry, arch and tag. This can be overwritten by arguments to make
-TAG?=1.1
+TAG?=1.2
 REGISTRY?=gcr.io/google_containers
+ARCH?=amd64
+
+CONTAINER_NAME_AND_TAG:=kube-dnsmasq-$(ARCH):$(TAG)
+
+KUBE_CROSS_IMAGE ?= gcr.io/google_containers/kube-cross
+KUBE_CROSS_VERSION ?= v1.6.2-2
+
+DNSMASQ_VERSION ?= dnsmasq-2.76
+
+ifeq ($(ARCH),amd64)
+	BASEIMAGE?=alpine:3.3
+	TRIPLE ?= x86_64-linux-gnu
+endif
+ifeq ($(ARCH),arm)
+	BASEIMAGE?=armel/busybox
+	TRIPLE ?= arm-linux-gnueabi
+endif
+ifeq ($(ARCH),arm64)
+	BASEIMAGE?=aarch64/busybox
+	TRIPLE ?= aarch64-linux-gnu
+endif
+ifeq ($(ARCH),ppc64le)
+	BASEIMAGE?=ppc64le/busybox
+	TRIPLE ?= powerpc64le-linux-gnu
+endif
 
 all: container
 
-container:
-	# And build the image
-	docker build -t $(REGISTRY)/dnsmasq:$(TAG) .
+sources:
+	@echo Downloading dnsmasq sources for cross compiled architectures
+	wget http://www.thekelleys.org.uk/dnsmasq/$(DNSMASQ_VERSION).tar.xz && tar xf $(DNSMASQ_VERSION).tar.xz
 
-push: container
+container.arm: sources container.cross
+
+container.arm64: sources container.cross
+
+container.ppc64le: sources container.cross
+
+container.cross: sources
+	# Build the dnsmasq binary using the cross-compilation docker image
+	docker run --detach=false --rm -v `pwd`/$(DNSMASQ_VERSION):/build $(KUBE_CROSS_IMAGE):$(KUBE_CROSS_VERSION) bash -c "cd /build; make clean; CC=$(TRIPLE)-gcc make"
```

Comment:
`--detach=false` => `-it`
consider merging `$(KUBE_CROSS_IMAGE):$(KUBE_CROSS_VERSION)` to only one variable, to make it less verbose